### PR TITLE
Added adapter for Conversant Media

### PR DESCRIPTION
**Pull request by msahagu2**
## Type of change
- [X] New bidder adapter
## Description of change

Added adapter for Conversant Media
- test parameters for validating bids

```
{
  bidder: 'conversant',
  params: {
    site_id: '87293'
  }
}
```
- contact email of the adapter’s maintainer: msahagun@conversantmedia.com
- [X] official adapter submission

### CODE REVIEW

File: src/adapters/conversant.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        responseCPM = parseFloat(conversantBid.price);
+
+        if (responseCPM !== 0.0) {
+          conversantBid.placementCode = placementCode;
+          conversantBid.size = bidRequested.sizes;
+          responseAd = conversantBid.adm || '';
+          responseNurl = conversantBid.nurl || '';
+
+          // Our bid!
+          bid = bidfactory.createBid(1);
```
Comment (protonate) [inline src/adapters/conversant.js]: pass the bid object as 2nd argument to sync bid request and response IDs. See #509

File: src/adapters/conversant.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+          } else {
+            bid.width = bidRequested.sizes[0][0];
+            bid.height = bidRequested.sizes[0][1];
+          }
+
+          bidmanager.addBidResponse(placementCode, bid);
+        } else {
+          //0 price bid
+          //indicate that there is no bid for this placement
+          bid = bidfactory.createBid(2);
```
Comment (protonate) [inline src/adapters/conversant.js]: pass the bid object as 2nd argument to sync bid request and response IDs. See #509

**Comment by mkendall07** [comment]
@msahagu2 
Please submit a docs PR here: https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders

File: src/adapters/conversant.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+      conversantImps.push(imp);
+    });
+
+    conversantBidReqs = {
+      'id': utils.getUniqueIdentifierStr(),
+      'imp': conversantImps,
+
+      'site': {
+        'id': siteId,
+        'mobile': document.querySelector('meta[name="viewport"][content*="width=device-width"]') !== null ? 1 : 0,
```
Comment (combs) [inline src/adapters/conversant.js]: is the presence of this meta tag a reliable way to detect mobile clients? Our responsive site at washingtonian.com includes this meta tag for everyone.

Comment (msahagu2) [inline src/adapters/conversant.js]: The mobile field is there to determine if the site's layout can be optimized when on mobile clients. Not necessarily that they are mobile

Comment (combs) [inline src/adapters/conversant.js]: I see, thank you @msahagu2! 👍

File: src/adapters/conversant.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+/**
+ * Adapter for requesting bids from Conversant
+ */
+var ConversantAdapter = function () {
+  var w = window,
+    n = navigator;
+
+  // production endpoint
+  var conversantUrl = '//media.msg.dotomi.com/s2s/header?callback=$$PREBID_GLOBAL$$.conversantResponse';
```
Comment (msahagu2) [inline src/adapters/conversant.js]: Do you know what ad sizes are requested? We're currently working on a fix for the headers of our 204 responses

Comment (combs) [inline src/adapters/conversant.js]: I am told by T. Rex at Conversant that the site_id numeric value must be passed as a string (buh?) so can you catch this case in the adapter @msahagu2

**Pull request merged**